### ISSUE TITLE
Uncaught exceptions when using maps of tuples

### ISSUE THREAD

**Issue by fubuloubu** [vyperlang/vyper#1432]
### Version Information

* vyper Version (output of `vyper --version`): 0.1.0b9
* OS: linux
* Python Version (output of `python --version`): Python 3.6.6
* Environment (output of `pip freeze`):
```
flake8==3.5.0
mccabe==0.6.1
pycodestyle==2.3.1
pycryptodome==3.6.6
pyflakes==1.6.0
```

### What's your issue about?
I got some uncaught exceptions when trying to work with maps of tuples:

```python
foo: map(uint256, (uint256, uint256))  # Works fine by itself
```

```python
@public
def bar1(_t: uint256):
    assert self.foo[_t] == (0, 0)
    # AttributeError: 'TupleType' object has no attribute 'unit'
```

```python
@public
def bar2(_t: uint256):
    self.foo[_t] = (1, 1)
    # UnboundLocalError: local variable 'subtype' referenced before assignment
```

```python
@public
@constant
def baz(_t: uint256) -> (uint256, uint256):
    return self.foo[_t]
    # UnboundLocalError: local variable 'subtype' referenced before assignment
```

### How can it be fixed?
Could be related to #1431.

**Comment by jacqueswww** [comment]
Same as in #1431 we don't support tuples as data types. And I don't think we should.

**Comment by fubuloubu** [comment]
Resolution is the same as it is in #1431 then: make it clearer that Tuples are not supported up front.

**Comment by iamdefinitelyahuman** [comment]
Fixed by #2029
